### DISCUSSION TITLE
Installing wz_mini_hacks to a Wyze V3 (instead of running from SD card)?

### DISCUSSION THREAD

**Discussion by supercoopers** [gtxaspec/wz_mini_hacks#761] [Q&A]
Hi, is there a way/instructions on how to run wz_mini_hacks directly on the camera without needing the SD card?  Maybe installing it from the SD card to the camera's memory?

**Comment by endertable** [comment]
The short answer is mostly a "no", if you are looking to keep all the features wz_mini has to offer.  The biggest reason is because of total size of the package, over 100M, will simply not fit on the built in memory chip of the cams.   Yes, you can scrape off the files not needed for a particular cam version or services you will not use, but possibly at the cost of other features.

There are a few things you can do to omit an SD card, but do note, they require modifying the cams firmware:

1. You could overwrite the kernel partition with a custom kernel so it kicks off your custom scripting.
2. Modify the rcS script so it calls your custom scripting.

These are just simply ways to not use the SD card for bootup, but at the end of the day, if you want anything useful from a hack, you are going to need an SD card for extra support files, e.g. libraries, scripts, binaries, modules, etc., unless you are utilizing Wyze's own endpoint stuff.

I guess the big questions are:  What is it you are trying to accomplish that makes you not want an SD card?  And what features do you want to keep from a cardless wz_mini?

**Reply by supercoopers** [reply]
thanks for the input! 

i've been using the wyze cam (purely in usb-webcam mode) in a 3d printed enclosure as a wearable/necklace camera for livestreams 

my idea was trying to remove the top pcb with the night vision LEDs and the sd card slot to make the camera thinner, as well as allowing for mounting wider angle/fisheye lenses that are currently too wide to fit through the cutout in the pcb.

- to do this my thought was to investigate installing webcam mode directly on the wyze cam, or somehow move the micro sd slot off that top pcb? (or maybe you guys have some good ideas?)

**Reply by virmaior** [reply]
At that point, why bother rejigging a wyze camera? is there really no product or set of parts that would do what you want?

**Reply by supercoopers** [reply]
I haven't found any USB camera modules yet with asgood night vision, local availability, and at a compatible price as the wyze cams! (But I've started looking into the sony imx modules)

**Reply by endertable** [reply]
What you are wanting to do, load custom firmware without an SD card, will require you modifying the original firmware.  But as already warned has potential for major implications. But if it is something you’re willing to go through then no prob. You have a few free partitions on the camera to be able to use, about four megs, to load just the basic USB modules.  This should be more than enough for what you are wanting to do.  I believe the easiest way to go about this would be to modify the rcS file to mount those partitions and run a custom script on them. 
This is not for the squeamish, but considering what you have already done, and what you are going to do hardware wise, this should be a walk in the park.  Just make sure your custom script has contingency functions until you are fully tested. The only thing I am not sure of is whether wz_mini’s custom kernel will be necessary or just being able to load the USB modules using insmod will be enough?!?

**Comment by virmaior** [comment]
along with what endertable is saying, I feel like there's also the ease of debugging issues with the SD card system.

Once you start sticking things into the camera's firmware itself, you're going to be completely screwed if you make a single mistake. At best you'll be able to UART and flash something else.

**Comment by gtxaspec** [comment]
The architecture of wzmini isn't really designed to be installed on to the flash.  If you are looking to install directly into the flash I would recommend alternative firmware such thingino, which is designed to be installed, though you lose the wyze ecosystem in doing so.
